### PR TITLE
Auth Cookie Support

### PR DESCRIPTION
**Pull request by yungcomputerchair**
Currently, an auth cookie login functions very similarly to a regular login, with the only difference being that the user's password is sent unencrypted through the auth ID field of the packet as a C-string instead of being encoded in the password field as Unicode. After extracting the password, the login flow is the same between the two.

This PR implements a proper cookie login flow, using one-shot text cookies for auth:
- Add an `Auth` table to the database that stores text cookies and their expiration timestamp
  - Cookies are stored in this table out-of-band, by something like an API server
- When a user performs a web login on a cookie-enabled server*, check that the cookie is correct and valid; allow login if so
- Invalidate the cookie for an account after any cookie login attempt

\* This new behavior is gated by an `authmethods` config option (which defaults to only allowing password login) so this will not break the auto-login trick that leverages the current behavior on servers that don't support real cookies.

### CODE REVIEW

**Comment by yungcomputerchair** [comment]
Builds are failing due to deprecated GitHub Actions workflows, not my code :P

File: src/servers/CNLoginServer.cpp [hunk 1]
```diff
@@ -638,4 +679,18 @@ bool CNLoginServer::isCharacterNameGood(std::string Firstname, std::string Lastn
     std::regex lastnamecheck(R"(((?! )(?!\.)[a-zA-Z0-9]*\.{0,1}(?!\.+ +)[a-zA-Z0-9]* {0,1}(?! +))*$)");
     return (std::regex_match(Firstname, firstnamecheck) && std::regex_match(Lastname, lastnamecheck));
 }
+
+bool CNLoginServer::isLoginTypeAllowed(LoginType loginType) {
+    // the config file specifies "comma-separated" but tbh we don't care
+    switch (loginType)
+    {
```
Comment (dongresource) [inline src/servers/CNLoginServer.cpp:686]: nit: Brace style. Not that it's ever been consistent in this codebase.

File: src/db/login.cpp [hunk 2]
```diff
@@ -98,6 +98,54 @@ void Database::updateAccountLevel(int accountId, int accountLevel) {
...
+        return false;
+    }
+
+    const char *cookie = reinterpret_cast<const char*>(sqlite3_column_text(stmt, 0));
+    if (strlen(cookie) != strlen(tryCookie)) {
+        sqlite3_finalize(stmt);
+        return false;
+    }
+
+    /* since cookies are immediately invalidated, we don't need to be concerned about
```
Comment (dongresource) [inline src/db/login.cpp:133]: nit: Block comment style: missing opening line.

**Review by dongresource — APPROVED** [review]
LGTM

**Pull request merged**